src/videojs-hls.js

```diff
@@ -881,7 +888,26 @@ videojs.Hls.getPlaylistDuration = function(playlist, startIndex, endIndex) {
 
   for (; i >= startIndex; i--) {
     segment = playlist.segments[i];
-    dur += (segment.duration !== undefined ? segment.duration : playlist.targetDuration) || 0;
+    if (segment.preciseDuration || segment.preciseTimestamp) {
```

Comment:
Can you calculate the segment duration immediately following the muxing so you don't have to check whether it needs to be calculated or not every time the duration is computed?